src/ssl/support.cc

```diff
@@ -192,6 +193,36 @@ int Ssl::asn1timeToString(ASN1_TIME *tm, char *buf, int len)
     return write;
 }
 
+// Compare 2 IP addresses
+static int compare_ip_addresses(void *check_data, ASN1_OCTET_STRING *altname_ip) {
+    // Cast check_data to const char
+    const unsigned char* check_ip = reinterpret_cast<const unsigned char*>(check_data);
+    sockaddr_storage socket_info;
+    if (!Ssl::is_ip_address(check_ip, &socket_info)) {
+        // We couldn't convert check_ip into either IPv4 or IPv6
+        debugs(83, 4, "Check data was not an IP address, not comparing..");
+        return -1;
+    }
+    if (socket_info.ss_family == AF_INET) {
+        // IPv4 conversion was successful, use it for comparison
+        debugs(83, 4, "IPV4 address found, checking..");
+        sockaddr_in* addr4 = reinterpret_cast<sockaddr_in*>(&socket_info);
+        if (altname_ip->length == sizeof(addr4->sin_addr) &&
+            std::memcmp(altname_ip->data, &(addr4->sin_addr), sizeof(addr4->sin_addr)) == 0) {
+            debugs(83, 4, "IPV4 comparison successful!");
+            return 0;
+        }
+    } else if (socket_info.ss_family == AF_INET6) {
+        debugs(83, 4, "IPV6 address found, checking..");
+        sockaddr_in6* addr6 = reinterpret_cast<sockaddr_in6*>(&socket_info);
+        if (altname_ip->length == sizeof(addr6->sin6_addr) &&
+            std::memcmp(altname_ip->data, &(addr6->sin6_addr), sizeof(addr6->sin6_addr)) == 0) {
```

Comment:
I hope that after Ip::Address::operator "=" is used to parse check_ip, we would be able to use existing Ip::Address comparison methods instead of raw memcmp(). Again, that existing code has many problems, but we do not want to semi-duplicate it unless we really have to.

Reply:
Yes, I've completely removed this function now and used functionality from Ip::Address. Thanks!